static/js/src/cube/components/MicrocertificationsTable/MicrocertificationsTable.tsx

```diff
@@ -1,109 +1,245 @@
+    studyLabURL: string,
+    takeURL: string,
+    status: Status
+  ) => (
+    <div className="u-no-padding--top u-align--right">
+      {status === Status.Enrolled ? (
+        <>
+          <a
+            className="p-button--neutral u-no-margin--right"
+            href={studyLabURL}
+          >
+            Prepare
+          </a>
+          <a className="p-button--positive" href={takeURL}>
+            Take
+          </a>
+        </>
+      ) : status === Status.InProgress ? (
+        <a className="p-button--positive" href={takeURL}>
+          Resume
+        </a>
+      ) : status === Status.Passed ? (
+        <ContextualMenu
+          hasToggleIcon
+          toggleLabel="Share"
+          links={[
             {
-              content: "Review and record relevant system information",
-              "aria-label": "Topics",
+              children: "Share via LinkedIn",
+              element: "a",
+              href: `https://www.linkedin.com/shareArticle?mini=true&amp;url=${badgeURL}`,
             },
             {
-              content: "Not Enrolled",
-              className: "p-table__cell--icon-placeholder",
-              "aria-label": "status",
+              children: "Share via Twitter",
+              element: "a",
+              href:
+                `https://twitter.com/share?text=I earned ${moduleName} ` +
+                `from @Canonical. Now I’m one step closer to becoming a ` +
+                `Certified @Ubuntu Engineer. Learn more about ` +
+                `Canonical’s available microcerts and CUBE 2020 at ` +
+                `https://ubuntu.com/cube.&amp;url=${badgeURL}&amp;` +
+                `hashtags=CUBE2020`,
             },
             {
-              content: <Button appearance={"positive"}>Purchase</Button>,
-              className: "u-align--right",
+              children: "Copy to clipboard",
+              onClick: () => copyBadgeUrl(badgeURL),
             },
-          ],
-        },
+          ]}
+        />
+      ) : (
+        <Button appearance={"positive"}>Purchase</Button>
+      )}
+    </div>
+  );
+
+  return (
+    <MainTable
+      responsive
+      className="p-table--cube--grid"
+      headers={[
+        { content: "#" },
+        { content: "" },
+        { content: "Module" },
+        { content: "Topics" },
+        { content: "Status", className: "p-table__cell--icon-placeholder" },
+        { content: "Action", className: "u-align--right" },
       ]}
+      rows={modules.map(
+        ({ name, badgeURL, topics, studyLabURL, takeURL, status }, index) => {
+          return {
+            key: index + 1,
+            columns: [
+              {
+                content: <span className="u-text--muted">{index + 1}</span>,
+                "aria-label": "Module number",
+              },
+              {
+                content: <img src={badgeURL} alt="" />,
+                className: "p-table--cube--grid__module-logo",
+                "aria-label": "Badge",
```

Comment:
But then again maybe it's fine, the module name is still there right below it. In this context, empty alt tag probably works. 👍

Reply:
Do you think it would be better to remove the `aria-label` here altogether? I don't think there's any useful information we could add to alt. As you said, I think it should be clear from the context, but on the other hand I don't rely on a screenreader and am not sure what would be the best experience for those that do.